src/ui/components/UploadScreen/UploadScreen.tsx

```diff
@@ -55,199 +53,41 @@ function DeletedScreen({ url }: { url: string }) {
-  isPublic: boolean;
-  setIsPublic: Dispatch<SetStateAction<boolean>>;
-}) {
-  const updateDefaultWorkspace = hooks.useUpdateDefaultWorkspace();
-  const selectedWorkspace = workspaces.find(w => w.id === selectedWorkspaceId);
-  const privateText =
-    selectedWorkspaceId === null ? (
-      `Only you can view this Replay`
-    ) : (
-      <span>
-        This replay can be viewed by anyone from{" "}
-        <span className="font-semibold">{selectedWorkspace!.name}</span>
-      </span>
-    );
-  const handleWorkspaceSelect = (id: string | null) => {
-    updateDefaultWorkspace({
-      variables: { workspaceId: id },
-    });
-    setSelectedWorkspaceId(id);
-  };
-
-  return (
-    <>
-      {workspaces.length ? (
-        <div className="">
-          <label className="block text-xs uppercase font-semibold">Team</label>
-          <TeamSelect {...{ workspaces, handleWorkspaceSelect, selectedWorkspaceId }} />
-        </div>
-      ) : null}
-      <div className="space-y-2 text-sm">
-        <label className="block text-xs uppercase font-semibold">Privacy</label>
-        <div className="space-y-1">
-          <div className="space-x-2 items-center">
-            <input
-              type="radio"
-              name="privacy"
-              value="public"
-              id="public"
-              checked={isPublic}
-              onChange={() => setIsPublic(true)}
-            />
-            <label htmlFor="public">
-              <span className="font-semibold">Public: </span>This replay can be viewed by anyone
-              with the link
-            </label>
-          </div>
-          <div className="space-x-2 items-center">
-            <input
-              type="radio"
-              name="privacy"
-              value="private"
-              id="private"
-              checked={!isPublic}
-              onChange={() => setIsPublic(false)}
-            />
-            <label htmlFor="private">
-              <span className="font-semibold">Private: </span>
-              {privateText}
-            </label>
-          </div>
-        </div>
-      </div>
-    </>
-  );
-}
-
-function SharingPreview({
-  toggleShowSharingSettings,
-  isPublic,
-  workspaces,
-  selectedWorkspaceId,
-}: {
-  toggleShowSharingSettings: () => void;
-  isPublic: boolean;
-  workspaces: Workspace[];
-  selectedWorkspaceId: string | null;
-}) {
-  const personalWorkspace = { id: null, name: "My Library" };
-  const workspaceName = [...workspaces, personalWorkspace].find(w => w.id === selectedWorkspaceId)!
-    .name;
-  let icon;
-  let text;
-
-  if (!isPublic) {
-    if (selectedWorkspaceId) {
-      text = `Shared privately with ${workspaceName}`;
-      icon = "groups";
-    } else {
-      text = `Only you can view this`;
-      icon = "person";
-    }
-  } else {
-    text = "This replay can be viewed by anyone with the link";
-    icon = "public";
-  }
-
+function ReplayScreenshot({ screenData }: { screenData: string }) {
```

Comment:
Huh... this got a lot simpler! I like that you've pushed this code off into discrete, meaningful places 👍